api/app/scripts/optimize_hyperparameters.py

```
@@ -0,0 +1,466 @@
+class HyperparamConfig:
+    """Configuration for a single hyperparameter trial."""
+
+    semantic_weight: float
+    keyword_weight: float
+    colbert_top_n: int
+    enable_colbert: bool
+
+    def __post_init__(self):
+        # Ensure weights sum to 1.0
+        if abs(self.semantic_weight + self.keyword_weight - 1.0) > 0.001:
+            raise ValueError("Weights must sum to 1.0")
```

Ruff (0.14.14)

Avoid specifying long messages outside the exception class

(TRY003)